### DISCUSSION TITLE
Z-Wave SmartStart provisioning (inclusion with QR code via S2 SmartStart)?

### DISCUSSION THREAD

**Discussion by Hedda** [zwave-js/zwave-js#3381] [Q&A]
Any development news around feature request to add Z-Wave SmartStart QR code provisioning now that Z-Wave JS support Security S2?

For those unaware; some Z-Wave Plus (Gen5) devices with S2 support devices already support starting Z-Wave SmartStart (a.k.a. Zwave Smart Start) QR code provisioning as an inclusion/pairing/joining beginning feature via QR codes printed on the product, and I understand that all Z-Wave Plus V2 devices (Z-Wave 700 series / Z-Wave Gen7 products) is mandatory required by certification to support Z-Wave SmartStart QR code provisioning (QR code as a method for provisioning to begin inclusion/join new devices). So while Z-Wave SmartStart QR codes is only available on a few select Z-Ware 500 series device today ("Z-Wave S2 certified") it will be standard on all existing and future Z-Wave 700 series devices that are "Z-Wave Plus V2" certified.

https://z-wavealliance.org/smartstart/

https://www.engineering.com/story/simplifying-z-wave-setup-with-smartstart

https://www.silabs.com/documents/public/white-papers/introduction_to_z-wave_smartstart_091317.pdf

https://www.aeotec.com/aeotec.com/z-wave-home-automation/z-wave-smart-start.html

https://qubino.com/smartstart-making-z-wave-both-useful-and-easy-to-set-up/

https://manuals.fibaro.com/document/adding-to-z-wave-network-5/

Also, do you think Home Assistant's Z-Wave JS integration maybe reuse some of the qr_code code from ZHA integration permit service? See:

https://community.home-assistant.io/t/request-qr-code-tags-scanning-for-joining-new-zigbee-3-0-device-and-z-wave-plus-v2-to-home-assistant-zha-and-z-wave-js-integration-component-via-ui-and-companion-apps/231185

https://www.home-assistant.io/integrations/zha#service-zhapermit

PS: HA's ZHA integration backend already supports qr_code parameter as input but there is no frontend/app scanning and parsing as of yet.

**Comment by AlCalzone** [comment]
Its on my list. Right now I'm busy with updating the application I originally created this library for. It has fallen quite behind the 3rd party applications in terms of feature parity.

**Comment by Hedda** [comment]
Given any more thoughts to Z-Wave SmartStart inclusion with QR code device installation via S2 SmartStart?

**Reply by ivdimitro** [reply]
Hi, I think this functionality is already implemented in the zwaveJS2mqtt GUI. If you go NodeManager and include/replace there is option to use S2 with QR code. It even works as I recently migrated some of my devices to S2 secure protocol.

**Reply by AlCalzone** [reply]
👆🏻 This :)

**Reply by Hedda** [reply]
> I think this functionality is already implemented in the zwaveJS2mqtt GUI. If you go NodeManager and include/replace there is option to use S2 with QR code. It even works as I recently migrated some of my devices to S2 secure protocol.

Great! Would then be awesome if QR-code scanning for Z-Wave SmartStart would also be implemented in third-party integrations.

Love it if QR-code scanning for Z-Wave SmartStart was implemented into Home Assistant's UI and its companion apps.

QR-code scanning (as well as barcode scanning) for Z-Wave and Zigbee device inclusion has previously been requested for HA:

https://community.home-assistant.io/t/request-qr-code-tags-scanning-via-ui-and-companion-apps-for-adding-new-zigbee-3-0-device-and-z-wave-plus-v2-to-home-assistant-zha-and-z-wave-js-integration-components/231185

https://community.home-assistant.io/t/barcode-scanner-built-into-companion-apps/343019/

**Reply by Hedda** [reply]
Looks like @raman325 has started working on adding SmartStart provisioning support to zwave_js WS API in Home Assistant core:

https://github.com/home-assistant/core/pull/59037

Nice!

**Reply by raman325** [reply]
> Looks like @raman325 has started working on adding SmartStart provisioning support to zwave_js WS API in Home Assistant core:
> 
> 
> 
> https://github.com/home-assistant/core/pull/59037
> 
> 
> 
> Nice! 

Backend is done at this point. Only reason why it's in draft is so that the front end can be built and we can confirm the model, but any potential changes required would be minimal.
